plugin/adsk/plugin/CMakeLists.txt

```diff
@@ -1,50 +1,99 @@
-set(PLUGIN_PACKAGE mayaUsdPlugin)
+#
+# Copyright 2019 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
+set(TARGET_NAME mayaUsdPlugin)
+
+#==============================================================================
+# packages
+#==============================================================================
 if(CMAKE_WANT_UFE_BUILD)
     find_package(UFE QUIET)
     if(UFE_FOUND)
         message(STATUS "Building with UFE ${UFE_VERSION} features enabled.")
-        include_directories(${UFE_INCLUDE_DIR})
-        add_definitions(-DWANT_UFE_BUILD)
     else()
         message(STATUS "UFE not found. UFE features will be disabled.")
     endif()
 endif()
 
-add_library(${PLUGIN_PACKAGE} MODULE
-    plugin.cpp
-    ProxyShape.cpp
+#==============================================================================
+# target
+#==============================================================================
+add_library(${TARGET_NAME} MODULE)
+
+# -----------------------------------------------------------------------------
+# sources
+# -----------------------------------------------------------------------------
+target_sources(${TARGET_NAME} 
+    PRIVATE
+        plugin.cpp
+        ProxyShape.cpp
 )
 
-target_compile_definitions(${PLUGIN_PACKAGE}
+# -----------------------------------------------------------------------------
+# compile definitions
+# -----------------------------------------------------------------------------
+target_compile_definitions(${TARGET_NAME}
     PRIVATE
         MAYAUSD_PLUGIN_EXPORT
 )
 
-target_include_directories(${PLUGIN_PACKAGE}
+if(UFE_FOUND)
+    target_compile_definitions(${TARGET_NAME} 
+        PRIVATE
+            -DWANT_UFE_BUILD
```

Comment:
I'm thinking that rather than have this in a separate if, it could be done using a "generator expression". Something like this just after line 50:

```cmake
$<UFE_FOUND:WANT_UFE_BUILD>
```

Note: you also don't need the -D as it will be removed (as per the docs).

Note: You can also use the same syntax for the target_include_directories below.

Reply:
@seando-adsk  Cool! Thanks for pointing it out. Even better! will change it.